### DISCUSSION TITLE
how to change color of error message

### DISCUSSION THREAD

**Discussion by ExpandingMan** [davatorium/rofi#1830] [Theming] [answered]
I for the life of me can't seem to figure out how to set the color of the text that appears when I type an argument into the prompt that results in an error (e.g. the name of a program that doesn't exist).

My config can be found [here](https://gitlab.com/ExpandingMan/dotfiles/-/blob/master/home/.config/rofi/config.rasi).

Anyone know how I can do this? Thanks!

**Comment by DaveDavenport** [comment]
See manpage: https://davatorium.github.io/rofi/1.7.5/rofi-theme.5/#error-message-structure

**Comment by ExpandingMan** [comment]
Yes, I have seen that, but I can't seem to figure out how to use it.  I have tried
```
error-message {
    text-color: @drac-red; 
}
```
and the same for `x.error-message` for various values of `x` but none of them seems to work.  I also can't seem to find `error` or `error-message` in any of the many examples available.

**Reply by DaveDavenport** [reply, marked as answer]
You are changing the color of the error-box, not the textbox that displays text..

```
rofi  -theme-str 'textbox { text-color: red; }'  -show drun
```

or 

```css
error-message {
text-color: red;
}
textbox {
text-color: inherit;
}
```

to test:

```bash
rofi  -theme-str 'error-message { text-color: red; } textbox { text-color: inherit;} '  -show drun
```

**Comment by ExpandingMan** [comment]
Ah, I see, thanks!

Out of curiosity, why does
```
error-message.textbox {
    text-color: @drac-red;
}
```
not work?

**Reply by DaveDavenport** [reply]
Why would it work?

**Reply by ExpandingMan** [reply]
I was envisioning this configuration as a tree such that branches can be specified with a `.` separated path, but I guess that is not what is going on.  Anyway, doesn't matter, I've been able to fix the one thing that was really broken with my configuration.

**Reply by DaveDavenport** [reply]
My first alpha used to be fully hierarchical but people complained it was to complex and hard to grasp. Therefor it was removed.

Please check the documentation for the current syntax,  I tried to make it complete.
